pil-analyzer/src/type_unifier.rs

```diff
@@ -41,20 +41,25 @@ impl Unifier {
         })
         .unwrap_or(ty);
 
-        if let Type::TypeVar(n) = ty {
-            self.add_type_var_bound(n.clone(), bound);
-        } else if let Type::NamedType(n) = ty {
-            // TODO Change this as soon as we support user-implemented traits.
-            return Err(format!("Type {n} does not satisfy trait {bound}."));
-        } else if bound == "ToString" && matches!(ty, Type::Array(_) | Type::Tuple(_)) {
-            // TODO Change this to a proper trait impl later.
-            for c in ty.clone().children().collect::<Vec<_>>() {
-                self.ensure_bound(c, "ToString".to_string())?;
+        match ty {
+            Type::TypeVar(n) => {
+                self.add_type_var_bound(n.clone(), bound);
             }
-        } else {
-            let bounds = elementary_type_bounds(ty);
-            if !bounds.contains(&bound.as_str()) {
-                return Err(format!("Type {ty} does not satisfy trait {bound}."));
+            Type::Array(_) | Type::Tuple(_) if bound == "ToString" => {
```

Comment:
I'd be nice if these were Rust types at some point instead of strings

Reply:
I think they will remain strings (or maybe SymbolPaths), but they will point to traits defined in the stdlib.